core/src/test/java/org/renjin/sexp/AttributeTest.java

```diff
@@ -292,7 +292,7 @@ public void setEmptyDimNames() {
   public void setEmptyDimNamesViaAttr() {
     eval("x <- 1:12");
     eval("dim(x) <- 3:4");
-    eval("attr(x, 'dimnames') <- list()");
+    eval("attr(x, 'dimnames') <- NULL");

```

Comment:
This is still important to check... if you assign an empty list to the dimnames attribute, the attribute _should_ be removed. (and absent attribute lookups always return `NULL`). I think the regression i introduced in the last commit needs to be fixed.